### ISSUE TITLE
Sections of a single transect associated to two different transect grid_ref

### ISSUE THREAD

**Issue by xaviermestdagh** [BiologicalRecordsCentre/ABLE#166]
Sections of the transect EBMS:Luxembourg:89 have two different grid_ref
S1 and S6 =>73863,135670
S3, S4, S5, S2 => 73881,135692
grid_ref should be unique for all sections of a single transect

**Comment by Gary-van-Breda** [comment]
This is all to do with the calculations used to work out the sref/centroid of the section. There are currently 3 options:

- Same as parent transect
- 100m grid square covering the centroid of the section
- 100m grid square covering the start of the section
  
The last 2 are only available if using WGC84, or OSGB / OSIE, otherwise it defaults to "Same as parent transect". In this case we are using LUREF, so it defaults to "Same as parent transect".  
As an aside, these last 2 do not work as advertised on WGC84 either, as the calculation rounds the Lat & Long to 0.001 degrees - which will vary for longitude depending on the latitude: at the equator is ~111m x ~111m, reducing to ~111m x ~62m at UK latitudes. The centroid is the centre point of the rectangle, as WGC describes a point, not an area.
  
Anyway, my plan is to add an additional option "1m grid square covering the centroid of the section", which would do a more accurate version of the above (0.00001 degrees for WGC84), with extra coding for LUREF.

**Comment by Gary-van-Breda** [comment]
Code deployed to Live.  
All section routes on EBMS:Luxembourg:89 re-saved, which has recalculated the centroid Sref for each, Now all showing different values.  
Any other sites with the same issue should have their routes re-saved as well.  
@xaviermestdagh : Is this OK?

**Comment by xaviermestdagh** [comment]
In the Full site section data download (csv), all sections from a single transect do have the same SRef coordinates, which is great. 
In the .shp export, sections of the transect EBMS:Luxembourg:89 still have different values of grid-ref.

**Comment by Gary-van-Breda** [comment]
OK I may have misunderstood: should the sections on the Transect all have the same Sref coordinate (which will be the same as the parent transect), or should they be different - i.e. be dependent on the route itself.

**Comment by xaviermestdagh** [comment]
Yes, misunderstanding from my side as well. 
I understood the field "SRef" means "Site ref" (=transect ref) in both csv exports (Full Site...csv). But if it means "Spatial Ref", so the "SRef" in the "Full Site Section Data Download.csv" should be the centroid of sections (no need in this file to have the centroid coordinates of the parent transect as it is accessible through a join on the "Site code").

**Comment by Gary-van-Breda** [comment]
Yes: in the CSV section download the first 6 columns (Site ID, Site Name, Site Code, Site Type, SRef, System) are all from the parent Site/Transect, not the section itself.
I don't have access to the SHP file view just now, but I would assume that is showing the Section data, not the Site/Transect.
Would you like me to change/add anything to either the SHP file data, or the CSV download?

**Comment by xaviermestdagh** [comment]
Having the section coordinates (of the centroid) in the .csv might avoid the need to calculate it in GIS from the .shp. 
In addition, I would recommend to rename the field "SRef" with a more self-understanding name (max 10 characters). For instance "SiteRef" or "TransRef".

**Comment by Gary-van-Breda** [comment]
CSV report update lodged into Git, ready to be merged into master branch and deployed to warehouse1 (@johnvanbreda)

**Comment by johnvanbreda** [comment]
Pulled onto the warehouse @Gary-van-Breda

**Comment by Gary-van-Breda** [comment]
Thanks @johnvanbreda   
@xaviermestdagh : I've double checked and the CSV Section download now has the site ref identified as such, plus an extra column for the Section Ref. This is used on the My sites and Scheme sites pages (and All sites, though I don't think you have access to that.)
